account_financial_risk/models/account_invoice.py

```diff
@@ -0,0 +1,106 @@
+# Copyright 2016-2018 Tecnativa - Carlos Dauden
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    risk_currency_id = fields.Many2one(related="partner_id.risk_currency_id")
+    risk_amount_total_currency = fields.Monetary(
+        string="Risk Amount Total",
+        currency_field="risk_currency_id",
+        compute="_compute_risk_amount_total_currency",
+    )
+
+    @api.depends(
+        "line_ids.debit",
+        "line_ids.credit",
+        "line_ids.currency_id",
+        "line_ids.amount_currency",
+        "line_ids.amount_residual",
+        "line_ids.amount_residual_currency",
+        "line_ids.payment_id.state",
+        "partner_id.credit_currency",
+        "partner_id.manual_credit_currency_id",
+        "partner_id.property_account_receivable_id.currency_id",
+        "partner_id.country_id",
+        "company_currency_id",
+    )
+    def _compute_risk_amount_total_currency(self):
+        for invoice in self:
+            invoice.risk_amount_total_currency = invoice.company_currency_id._convert(
+                invoice.amount_total_signed,
+                invoice.risk_currency_id,
+                invoice.company_id,
+                invoice.invoice_date or fields.Date.context_today(self),
+                round=False,
+            )
+
+    def risk_exception_msg(self):
+        self.ensure_one()
+        partner = self.partner_id.commercial_partner_id
+        exception_msg = ""
+        if partner.risk_exception:
+            exception_msg = _("Financial risk exceeded.\n")
```

Comment:
Please apply to all use cases. This is not required but it is for time-efficiency

```suggestion
            exception_msg = self.env._("Financial risk exceeded.\n")
```

Reply:
done